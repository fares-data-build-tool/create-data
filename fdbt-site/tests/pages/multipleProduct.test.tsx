/* eslint-disable @typescript-eslint/no-non-null-assertion */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallow } from 'enzyme';
import React from 'react';
import MultipleProducts, { getServerSideProps } from '../../src/pages/multipleProducts';
import { getMockContext } from '../testData/mockData';

describe('pages', () => {
    describe('multipleProduct', () => {
        it('should render correctly', () => {
            const wrapper = shallow(
                <MultipleProducts
                    numberOfProductsToDisplay="2"
                    nameOfOperator="Infinity Line"
                    passengerType="Adult"
                    errors={[]}
                    userInput={[]}
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });

        describe('getServerSideProps', () => {
            beforeEach(() => {
                jest.clearAllMocks();
            });
            it('should return number of products to display, name of operator and passenger type if there is no cookie set', () => {
                const ctx = getMockContext({
                    operator: 'BLP',
                    passengerType: { passengerType: 'Adult' },
                    numberOfProductsInput: '2',
                });
                const result = getServerSideProps(ctx);

                expect(result.props.numberOfProductsToDisplay).toBe('2');
                expect(result.props.nameOfOperator).toBe('BLP');
                expect(result.props.passengerType).toBe('Adult');
            });

            it('should throw an error if the necessary cookies to render the page are not present', () => {
                const ctx = getMockContext({
                    operator: null,
                    numberOfProductsInput: null,
                });
                expect(() => getServerSideProps(ctx)).toThrow(
                    'Necessary cookies not found to show multiple products page',
                );
            });
        });
    });
});
