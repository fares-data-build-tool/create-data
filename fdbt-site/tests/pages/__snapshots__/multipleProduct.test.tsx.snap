// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages multipleProduct should render correctly 1`] = `
<Layout
  description="Multiple Product entry page of the Fares Data Build Tool"
  title="Multiple Product - Fares Data Build Tool"
>
  <main
    className="govuk-main-wrapper app-main-class"
    id="main-content"
    role="main"
  >
    <form
      action="/api/multipleProducts"
      method="post"
    >
      <ErrorSummary
        errors={Array []}
      />
      <div
        className="govuk-form-group"
      >
        <fieldset
          aria-describedby="multiple-product-page-heading"
          className="govuk-fieldset"
        >
          <legend
            className="govuk-fieldset__legend govuk-fieldset__legend--xl"
          >
            <h1
              className="govuk-fieldset__heading"
              id="multiple-product-page-heading"
            >
              Enter your product details
            </h1>
          </legend>
          <span
            className="govuk-hint"
            id="service-operator-hint"
          >
            Infinity Line
             - 
            2
             Products - 
            Adult
          </span>
        </fieldset>
        <div
          className="govuk-inset-text"
        >
          For example, Super Saver ticket - Â£4.95 - 2
        </div>
      </div>
      <div
        className="govuk-grid-row"
      >
        <ProductRow
          errors={Array []}
          numberOfProductsToDisplay="2"
          userInput={Array []}
        />
      </div>
    </form>
  </main>
</Layout>
`;
